includes/core/server/request/class-get-reporting-payment-activity.php

```diff
@@ -50,32 +50,32 @@ public function get_method(): string {
 	/**
 	 * Sets the start date for the payment activity data.
 	 *
-	 * @param string|null $date_start The start date in the format 'YYYY-MM-DDT00:00:00' or null.
+	 * @param string $date_start The start date in the format 'YYYY-MM-DDT00:00:00' or null.
 	 * @return void
 	 */
-	public function set_date_start( ?string $date_start ) {
-		// TBD - validation.
+	public function set_date_start( string $date_start ) {
+		$this->validate_date( $date_start, 'Y-m-d\TH:i:s' );
 		$this->set_param( 'date_start', $date_start );
 	}
 
 	/**
 	 * Sets the end date for the payment activity data.
 	 *
-	 * @param string|null $date_end The end date in the format 'YYYY-MM-DDT00:00:00' or null.
+	 * @param string $date_end The end date in the format 'YYYY-MM-DDT00:00:00' or null.
 	 * @return void
 	 */
 	public function set_date_end( string $date_end ) {
-		// TBD - validation.
+		$this->validate_date( $date_end, 'Y-m-d\TH:i:s' );
 		$this->set_param( 'date_end', $date_end );
 	}
 
 	/**
 	 * Sets the timezone for the reporting data.
 	 *
-	 * @param string|null $timezone The timezone to set or null.
+	 * @param string $timezone The timezone to set or null.
 	 * @return void
 	 */
-	public function set_timezone( ?string $timezone ) {
-		$this->set_param( 'timezone', $timezone ?? 'UTC' );
+	public function set_timezone( string $timezone ) {
+		$this->set_param( 'timezone', $timezone );
```

Comment:
We can actually validate timezone as well by passing it to `DateTimeZone` constructor as it will return [an exception](https://www.php.net/manual/en/datetimezone.construct.php#refsect1-datetimezone.construct-errors) if string is not a timezone representation.

Feel free to add a validation here in the client or not. The server will return an error anyway if string is not a valid timezone.

Reply:
If you decide to add some validation for the timezone as well, you might need to update https://github.com/Automattic/woocommerce-payments/blob/fix/8742-merchant-timezone-payment-activity/includes/core/server/request/class-get-reporting-payment-activity.md#exceptions too.

Reply:
Added validation and updated docs.